src/v/storage/chunk_cache.cc

```diff
@@ -37,6 +39,36 @@ ss::future<> chunk_cache::start() {
       });
 }
 
+ss::future<> chunk_cache::stop() {
+    _metrics.clear();
+    return ss::now();
+}
+
+void chunk_cache::setup_metrics() {
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    namespace sm = ss::metrics;
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("chunk_cache"),
+      {
+        sm::make_gauge(
+          "total_size_bytes",
+          [this] { return _size_total; },
+          sm::description("Total size of the chunk cache in bytes")),
+        sm::make_gauge(
+          "available_size_bytes",
```

Comment:
maybe "free" is better? I used available in the context of the memory metrics only because "free" was taken

Reply:
I mean don't mind I was mostly just matching the existing terms being used. In this case it's a bit tricky because one could say that "free" is really `_size_available + _size_target - _size_total`.

Reply:
> I mean don't mind I was mostly just matching the existing terms being used.

Do you mean matching the terms here in the chunk cache code, or the terms in other memory metrics (like `vectorized_memory_available_memory`)?

In the other memory metrics "free" is the amount of memory actually unused and ready for immediate allocation, and "available" is the amount which could probably be make available after some work (reclaim)?

So along those lines I think "free" would be `_size_available` and "available" would be `_size_limit - _size_total`.

> `_size_available + _size_target - _size_total`

Not follow that math: if target (lwm) was 1G and no chunks were outstanding in the appenders wouldn't that give 1 + 1 + 0 = 2 GB?

Reply:
> Do you mean matching the terms here in the chunk cache code

Yeah was referring to the chunk cache code.

That math was garbage.